python/tvm/relay/frontend/pytorch.py

```diff
@@ -0,0 +1,1135 @@
+            node_name = [output.debugName() for output in node.outputs()][0]
+
+            if node.kind() == "prim::Constant":
+                node_value = '0'
+                if "None" not in node_str and "?" not in node_str:
+                    node_value = ((node_str.split(' = ')[1]).split('value=')[1]).split(']')[0]
+                self._consts[node_name] = node_value
+            elif node.kind() == "prim::ListConstruct":
+                list_shape = []
+                for input_node in node.inputs():
+                    if input_node.debugName() in self._inputs_r.keys():
+                        list_shape.append(int(self._inputs_r[input_node.debugName()]))
+                    elif input_node.debugName() in self._consts.keys():
+                        list_shape.append(int(self._consts[input_node.debugName()]))
+                self._inputs_r[node_name] = _expr.var(node_name, shape=list_shape)
+            elif node.kind() == "prim::GetAttr":
+                continue
+
+            self._add_op(node_name, node)
+
+    # Graph Helper Functions
+
+    def _add_op(self, node_id, op_node):
+        """ Add an operator and its operators inputs to the graph and insert placeholders
+            where an input is a call node.
+
+        Parameters
+        ----------
+        node_id : string
+            The ID of the op node
+
+        op_node : PyTorch Node object
+            The full Node object for the op node
+
+        """
+        self._ops[(node_id)] = op_node
+        input_list_r = []
+        input_list_types = []
+        for input_node in op_node.inputs():
+            if input_node.debugName() in self._inputs_r.keys():
+                input_list_r.append(self._inputs_r[input_node.debugName()])
+            elif input_node.debugName() in self._params.keys():
+                input_list_r.append(self._params[input_node.debugName()])
+            elif input_node.node().kind() == "prim::Constant":
+                input_list_r.append(self._consts[input_node.debugName()])
+            else:
+                input_list_r.append("call/var."+input_node.debugName())
+
+                # If the inputs of a ListConstruct op is a call or var, remove it from inputs
+                if op_node.kind() == 'prim::ListConstruct':
+                    if node_id in self._inputs_r.keys():
+                        self._inputs_r.pop(node_id)
+
+            try:
+                input_node_kind = input_node.type().kind()
+                if input_node_kind == 'TensorType':
+                    if input_node.type().scalarType() is None:
+                        input_list_types.append('float')
+                    else:
+                        input_list_types.append(input_node.type().scalarType().lower())
+                elif input_node_kind == 'ListType':
+                    input_list_types.append(str(input_node.type().getElementType()).lower())
+                elif input_node_kind == 'IntType' or input_node_kind == 'FloatType' or \
```

Comment:
needs cleanup
`elif input_node_kind in ['IntType', 'FloatType', 'BoolType', 'StringType', 'OptionalType']:`